newapi/ooniapi/citizenlab.py

```diff
@@ -750,13 +753,110 @@ def list_url_priorities():
     log.debug("listing URL prio rules")
     query = """SELECT category_code, cc, domain, url, priority
     FROM url_priorities
-    ORDER BY 2, 1, 3, 4
+    ORDER BY category_code, cc, domain, url, priority
     """
-    q = current_app.db_session.execute(query)
-    rows = [dict(r) for r in q]
+    if current_app.config["USE_CLICKHOUSE"]:
+        q = query_click(sql.text(query), {})
+        rows = list(q)
+    else:
+        q = current_app.db_session.execute(query)
+        rows = [dict(r) for r in q]
     return make_response(jsonify(rules=rows))
 
 
+def update_url_priority_postgresql(old, new):  # pragma: no cover
+    log = current_app.logger
+    if old:  # delete an existing rule
+        query = """DELETE FROM url_priorities
+        WHERE category_code = :category_code
+        AND cc = :cc
+        AND domain = :domain
+        AND url = :url
+        AND priority = :priority
+        """
+        q = current_app.db_session.execute(query, old).rowcount
+        if q < 1:
+            return jerror("Old rule not found", 400)
+
+    if new:  # add new rule
+        query = """INSERT INTO url_priorities
+            (category_code, cc, domain, url, priority)
+            VALUES(:category_code, :cc, :domain, :url, :priority)
+        """
+        try:
+            q = current_app.db_session.execute(query, new).rowcount
+        except Exception as e:
+            log.info(str(e))
+            current_app.db_session.rollback()
+            return jerror("Duplicate rule", 400)
+
+    current_app.db_session.commit()
+    return q
+
+
+def dicthash(d: dict):
+    return hash(tuple(sorted(d.items())))
+
+def init():
+    rules = [
+        ('NEWS', '*', '*', '*', 100),
+        ('POLR', '*', '*', '*', 100),
+        ('HUMR', '*', '*', '*', 100),
+        ('LGBT', '*', '*', '*', 100),
+        ('ANON', '*', '*', '*', 100),
+        ('MMED', '*', '*', '*', 80),
+        ('SRCH', '*', '*', '*', 80),
+        ('PUBH', '*', '*', '*', 80),
+        ('REL', '*', '*', '*', 60),
+        ('XED', '*', '*', '*', 60),
+        ('HOST', '*', '*', '*', 60),
+        ('ENV', '*', '*', '*', 60),
+        ('FILE', '*', '*', '*', 40),
+        ('CULTR', '*', '*', '*', 40),
+        ('IGO', '*', '*', '*', 40),
+        ('GOVT', '*', '*', '*', 40),
+        ('DATE', '*', '*', '*', 30),
+        ('HATE', '*', '*', '*', 30),
+        ('MILX', '*', '*', '*', 30),
+        ('PROV', '*', '*', '*', 30),
+        ('PORN', '*', '*', '*', 30),
+        ('GMB', '*', '*', '*', 30),
+        ('ALDR', '*', '*', '*', 30),
+        ('GAME', '*', '*', '*', 20),
+        ('MISC', '*', '*', '*', 20),
+        ('HACK', '*', '*', '*', 20),
+        ('ECON', '*', '*', '*', 20),
+        ('COMM', '*', '*', '*', 20),
+        ('CTRL', '*', '*', '*', 20),
+        ('NEWS', 'it', '*', '*', 10),
+        ('NEWS', 'it', 'www.leggo.it', '*', 5),
+        ('COMT', '*', '*', '*', 100),
+        ('GRP', '*', '*', '*', 100),
+    ]
```

Comment:
Why are these priorities hardcoded?

How will the updating of the priorities work? Will the prioritization frontend write to pg or clickhouse? How do we ensure the two tables are kept in sync? If it's not needed to keep them in sync, perhaps we should be dropping support for reading/writing to whichever one is no longer being used.

Reply:
This was used only once just to bootstrap default values in the table.